### ISSUE TITLE
Customize structured Access Token

### ISSUE THREAD

**Issue by ofairfoul** [panva/node-oidc-provider#520]
**Is your feature request related to a problem? Please describe.**
I would like to be able to customize the JWT Access Token format by providing my own formatter class. The reason is I need to add an x5t from my signing certificate to the header for compatibility with an old identity server.

**Describe the solution you'd like**
Add something to the confg object to add a custom formatter
```js
{
  customFormats: {
    'jwt-custom': MyCustomJwtFormatter
  },
  formats: {
    AccessToken: 'jwt-custom'
  }
}
```

**Describe alternatives you've considered**
An alternative would be some extension to the jwt formatter to include custom headers along with claims.

**Additional context**

 - [x] i have searched the configuration section for this feature and couldn't find it
 - [x] i have searched the issues tracker on github for similar requests and couldn't find anything related.

**Comment by davidgtonge** [comment]
This would be useful for us as well. At the moment we are overriding the AccessToken class

**Comment by panva** [comment]
@ofairfoul @davidgtonge can you elaborate on the customizations you wish to make so that I can think of a solution?

**Comment by davidgtonge** [comment]
We wanted to add some custom claims to the access token. I think at the moment we bundle all the claims associated with the grant in the access token.

**Comment by panva** [comment]
@davidgtonge in v6.x there's already [`extraAccessTokenClaims`](https://github.com/panva/node-oidc-provider/blob/master/docs/README.md#extraaccesstokenclaims) that's going to push everything non-conflicting to the structured access token.

**Comment by panva** [comment]
@ofairfoul please gives this a try and provide feedback.

```
npm i panva/node-oidc-provider#customize-structured-tokens
```

https://github.com/panva/node-oidc-provider/blob/customize-structured-tokens/docs/README.md#customizestructuredaccesstoken

**Comment by panva** [comment]
Released in [v6.7.0](https://github.com/panva/node-oidc-provider/releases/tag/v6.7.0)

**Comment by ofairfoul** [comment]
@panva thanks so much this is amazing work, it works perfectly. 
The only minor suggestion I would like to make is to maybe include the key (and alg?) in the customizer function signature. I know which key is likely to be chosen based on my config and sort of hardcode it it but think it would be less prone to error to use the same key resolver as the token format model.
